src/views/Home.jsx

```diff
@@ -14,11 +22,48 @@ export function Home({ createNewToken, setListToken }) {
 		navigate('/list');
 	};
 
+	const handleTokenChange = (e) => {
+		setExistingToken(e.target.value);
+	};
+
+	const handleSubmit = async (e) => {
+		e.preventDefault();
+		try {
+			const tokenExists = await getExistingList(existingToken);
+			if (tokenExists) {
+				setListToken(existingToken);
+				navigate('/list');
+			} else {
+				setStatus("That list doesn't exist! Create a new list to get started.");
+			}
+			// Clear the existingToken after the submission
+			setExistingToken('');
+		} catch (error) {
+			throw error;
+		}
+	};
+
 	return (
 		<div className="Home">
 			<p>
 				Hello from the home (<code>/</code>) page!
 			</p>
+			<form onSubmit={handleSubmit}>
+				<label htmlFor="existingToken">
+					Which list would you like to join?{' '}
```

Comment:
Not sure what this is for?
```suggestion
					Which list would you like to join?
```